src/Microsoft.Performance.Toolkit.Engine/Engine.cs

```diff
@@ -983,59 +983,146 @@ private RuntimeExecutionResults ProcessCore()
+            {
+                var processingSource = psDsgPair.Key.Item1;
+                var dsg = psDsgPair.Key.Item2;
+                var processorOptions = psDsgPair.Value;
+
+                try
+                {
+                    /*
+                     * TODO (#214): for now, the engine ignores the groups returned by processing sources that implement
+                     * IDataSourceGrouper. In the future, this needs to
+                     * 1. Check if the processing source implements IDataSourceGrouper (which it must after V2)
+                     * 2. If not, do what we're doing now and just create a new DataSourceGroup with the default
+                     *     processing mode
+                     * 3. If yes, use an IDataSourceGroupResolver to pick out which of the IDataSourceGroup(s)
+                     *    returned by the processing source should be used for processing.
+                     *      a. If the engine user does not supply a resolver, a sensible one should be used. E.g.
+                     *         one that picks groups that maximize the number of data sources processed without a
+                     *         data source being processed in more than 1 group
+                     * 4. Add a new executor for each IDataSourceGroups picked out in the above step
+                     */
+
+                    var executionContext = new SDK.Runtime.ExecutionContext(
+                        new DataProcessorProgress(),
+                        x => ConsoleLogger.Create(x.GetType()),
+                        processingSource,
+                        dsg, // todo #214
+                        processingSource.Instance.MetadataTables,
+                        new RuntimeProcessorEnvironment(this.Extensions, this.compositeCookers, this.CreateLogger),
+                        processorOptions);
+
+                    var executor = new ProcessingSourceExecutor();
+                    executor.InitializeCustomDataProcessor(executionContext);
+
+                    executors.Add(executor);
+                }
+                catch (Exception e)
+                {
+                    this.logger.Error(
+                       "Error processing dataSources {0} on processing source {1}: {2}",
+                       dsg.DataSources.ToString(),
+                       processingSource.Name,
+                       e);
                 }
             }
 
             return executors;
         }
 
+        private static bool AreUnsupportedOptions(
+            ProcessingSourceReference processingSourceReference,
+            ProcessorOptions processorOptions, 
+            ILogger logger)
+        {
+            bool areUnsupportedOptions = false;
+
+            if (processorOptions is null)
+            {
+                logger.Warn($"Provided Processor options are null for ${processingSourceReference} (${processingSourceReference.Guid}).");
+                return true;
+            }
```

Comment:
processorOptions should never be null, should they? if not, can we make this an assert as it's a private method?

Reply:
ah, nevermind. I missed something earlier.